src/main/java/org/codevillage/TypeParsingStep.java

```diff
@@ -0,0 +1,65 @@
+package org.codevillage;
+
+import com.github.javaparser.StaticJavaParser;
+import com.github.javaparser.ast.CompilationUnit;
+import com.github.javaparser.ast.body.TypeDeclaration;
+import com.github.javaparser.utils.SourceRoot;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Optional;
+
+public class TypeParsingStep extends EntityParsingStep {
+
+    public TypeParsingStep(EntityParsingChain next) {
+        super(next);
+    }
+
+    @Override
+    public JavaEntity construct(EntityBuilder builder, CompilationUnit declaration) {
+        List<TypeDeclaration<?>> types = declaration.getTypes();
+
+        if (!types.isEmpty()) {
+            TypeDeclaration<?> typeDeclaration = types.get(0); // assuming there's only one type in the file
```

Comment:
So there is a very decent chance that this first thing is not the ClassOrInterface declaration that we are looking for (could also be an enum or an annotation i think), so it might be easiest just search for the first member that is an `instanceof ClassOrInterfaceDeclaration` and then determine the type of that. Just because I think we might get errors otherwise.

Reply:
And we could maybe return null if it doesn't exist?